src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -21,150 +22,24 @@
-        self._backend.openssl_assert(issuer != self._backend._ffi.NULL)
-        return _decode_x509_name(self._backend, issuer)
-
-    @property
-    def subject(self) -> x509.Name:
-        subject = self._backend._lib.X509_get_subject_name(self._x509)
-        self._backend.openssl_assert(subject != self._backend._ffi.NULL)
-        return _decode_x509_name(self._backend, subject)
-
-    @property
-    def signature_hash_algorithm(
-        self,
-    ) -> typing.Optional[hashes.HashAlgorithm]:
-        oid = self.signature_algorithm_oid
-        try:
-            return x509._SIG_OIDS_TO_HASH[oid]
-        except KeyError:
-            raise UnsupportedAlgorithm(
-                "Signature algorithm OID:{} not recognized".format(oid)
-            )
-
-    @property
-    def signature_algorithm_oid(self) -> x509.ObjectIdentifier:
-        alg = self._backend._ffi.new("X509_ALGOR **")
-        self._backend._lib.X509_get0_signature(
-            self._backend._ffi.NULL, alg, self._x509
-        )
-        self._backend.openssl_assert(alg[0] != self._backend._ffi.NULL)
-        oid = _obj2txt(self._backend, alg[0].algorithm)
-        return x509.ObjectIdentifier(oid)
-
-    @utils.cached_property
-    def extensions(self) -> x509.Extensions:
-        return self._backend._certificate_extension_parser.parse(self._x509)
-
-    @property
-    def signature(self) -> bytes:
-        sig = self._backend._ffi.new("ASN1_BIT_STRING **")
-        self._backend._lib.X509_get0_signature(
-            sig, self._backend._ffi.NULL, self._x509
-        )
-        self._backend.openssl_assert(sig[0] != self._backend._ffi.NULL)
-        return _asn1_string_to_bytes(self._backend, sig[0])
-
-    @property
-    def tbs_certificate_bytes(self) -> bytes:
-        pp = self._backend._ffi.new("unsigned char **")
-        res = self._backend._lib.i2d_re_X509_tbs(self._x509, pp)
-        self._backend.openssl_assert(res > 0)
-        pp = self._backend._ffi.gc(
-            pp, lambda pointer: self._backend._lib.OPENSSL_free(pointer[0])
-        )
-        return self._backend._ffi.buffer(pp[0], res)[:]
-
-    def public_bytes(self, encoding: serialization.Encoding) -> bytes:
-        bio = self._backend._create_mem_bio_gc()
-        if encoding is serialization.Encoding.PEM:
-            res = self._backend._lib.PEM_write_bio_X509(bio, self._x509)
-        elif encoding is serialization.Encoding.DER:
-            res = self._backend._lib.i2d_X509_bio(bio, self._x509)
-        else:
-            raise TypeError("encoding must be an item from the Encoding enum")
-
-        self._backend.openssl_assert(res == 1)
-        return self._backend._read_mem_bio(bio)
+# This exists for pyOpenSSL compatibility and SHOULD NOT BE USED
+# WE WILL REMOVE THIS VERY SOON.
+def _Certificate(backend, x509) -> x509.Certificate:  # noqa: N802
+    warnings.warn(
+        "This version of cryptography contains a temporary pyOpenSSL "
+        "fallback path. Upgrade pyOpenSSL now.",
```

Comment:
Do we have a PR up on the pyOpenSSL side?

Reply:
Not yet, no.

Reply:
https://github.com/pyca/pyopenssl/pull/1030